guidance/library/_json.py

```diff
@@ -20,9 +20,11 @@
         raise
 
 from .._guidance import guidance
-from ..library import char_range, one_or_more, optional, zero_or_more
+from ..library import char_range, gen, one_or_more, optional, zero_or_more
 
 from .._grammar import GrammarFunction, select, capture, with_temperature
+from ._at_most_n_repeats import at_most_n_repeats
+from ._exactly_n_repeats import exactly_n_repeats
```

Comment:
Could be nice to use/reuse these to reduce redundant code in the regex implementation :)